src/components/TreeView/Tree.tsx

```diff
@@ -3,19 +3,20 @@ import { INode, INodeContainer, ITree } from './types';
 import { Node } from './Node';
 
 const level = 1;
-export function Tree<T extends INode>({ dataSource, nodeView, onNodeClicked, className, compareNodes, childrenProperty, getKey, onViewNodeDetails }: ITree<T>) {
+export function Tree<T extends INode>({ parent, dataSource, nodeView, onNodeClicked, className, compareNodes, childrenProperty, getKey, onViewNodeDetails, onTransferClick, onUnnestClick }: ITree<T>) {
   const [dataSourceState, setDataSourceState] = useState(dataSource);
 
   useEffect(() => {
     setDataSourceState(dataSource);
   }, [dataSource]);
 
   const nodeClickHandler = useCallback((data: T) => {
-    const selectNode = (node: any) => {
+    const selectNode = (node: T) => {
```

Comment:
you can use "INode" instead of "T" here (since it is already expected to be inherited from) -> you will not need to cast childrenProperty